lib/src/sql/datetime.rs

```diff
@@ -199,15 +200,16 @@ fn day(i: &str) -> IResult<&str, u32> {
 }
 
 fn hour(i: &str) -> IResult<&str, u32> {
-	take_digits_range(i, 2, 0..=24)
+	take_digits_range(i, 2, 0..=23)
 }
 
 fn minute(i: &str) -> IResult<&str, u32> {
 	take_digits_range(i, 2, 0..=59)
 }
 
 fn second(i: &str) -> IResult<&str, u32> {
-	take_digits_range(i, 2, 0..=59)
+	// 60 is for leap seconds
+	take_digits_range(i, 2, 0..=60)
```

Comment:
See [RFC 3339 §5.6](https://www.rfc-editor.org/rfc/rfc3339#section-5.6), I don't have access to ISO 8601, so this is the closest thing. 

@tobiemh you should buy [ISO 8601-1](https://www.iso.org/standard/70907.html) and [ISO 8601-2](https://www.iso.org/standard/70908.html) for me and I'll redo all this parsing to spec.